dagshub/common/api/repo.py

```diff
@@ -259,78 +260,134 @@ def get_storage_file(self, path: str) -> bytes:
+            remote_path: Path in the repository of the folder or file to download.
+            local_path: Where to download the files. Defaults to current working directory.
+            revision: Repo revision or branch, if not specified - uses default repo branch.
+                Ignored for downloading from buckets.
+            recursive: Whether to download files recursively.
+            keep_source_prefix: | Whether to keep the path of the folder in the download path or not.
+                | Example: Given remote_path ``src/data`` and file ``test/file.txt``
+                | if ``True``: will download to ``<local_path>/src/data/test/file.txt``
+                | if ``False``: will download to ``<local_path>/test/file.txt``
+            redownload: Whether to redownload files that already exist on the local filesystem.
+                The downloader doesn't do any hash comparisons and only checks
+                if a file already exists in the local filesystem or not.
+            download_storages: If downloading the whole repo, by default we're not downloading the integrated storages
+                Toggle this to ``True`` to change this behavior
+        """
+        traverse_storages = True
+        if str(remote_path) == "/" and not download_storages:
+            log_message(
+                "Skipping downloading from connected storages. "
+                "Set the `download_storages` flag if you want "
+                "to download the whole content of the connected storages."
+            )
+            traverse_storages = False
+
+        files = self._get_files_in_path(remote_path, revision, recursive, traverse_storages=traverse_storages)
         file_tuples = []
         if local_path is None:
             local_path = "."
         local_path = Path(local_path)
         # Strip the slashes from the beginning so the relative_to logic works
-        if str(remote_path.startswith("/")):
-            remote_path = str(remote_path).lstrip("/")
+        remote_path = str(remote_path).lstrip("/")
+        if not remote_path:
+            remote_path = "/"
+        # For storage paths get rid of the colon in the beginning of the schema, the download urls won't have it either
+        if remote_path.split("/")[0] in {"s3:", "gs:", "azure:"}:
+            remote_path = remote_path.replace(":", "", 1)
         # Edge case - if the user requested a single file - different output path semantics
         if len(files) == 1 and files[0].path == remote_path:
             f = files[0]
             remote_path = PurePosixPath(f.path)
             # If local_path was specified, assume that the local_path is the exact name of the file
             if local_path != Path("."):
-                file_path = local_path
+                # Saving to existing dir - append the name of remote file to the end a-la cp
+                if local_path.exists() and local_path.is_dir():
```

Comment:
If it exists and not a dir, you should throw.
Also, make sure this works with symlinks (in the middle and end of the paths)

Reply:
If it exists and not a dir, it's going to defer to the `redownload` flag. If it exists, it just won't be downloaded

Reply:
I don't think I ever had problems with pathlib and symlinks